Source/CarthageKit/Git.swift

```diff
@@ -394,6 +382,24 @@ private func parseConfigEntries(_ contents: String, keyPrefix: String = "", keyS
 	}
 }
 
+/// Git’s representation of file system objects at a path relative to the repository root.
+///
+/// - parameter pathTransformWhereNilResultsWillBeEliminated: Defaults to identity function. Only ever used for one thing, but parameterized for unit testing purposes.
+internal func fileSystemObjects(for dependency: Dependency<PinnedVersion>, withRepository repositoryURL: URL, at pathList: String, pathTransformWhereNilResultsWillBeEliminated: @escaping (String) -> String? = { $0 }) -> SignalProducer<[String], CarthageError> {
```

Comment:
> `for dependency: Dependency<PinnedVersion>`

Let's have this not take a `Dependency`—treat it as a pure wrapper for the git function.

> `at pathList: String`

This seems like it only takes a single path? Or at least is only used that way? It should also have a more explicit external name to compensate for the `String` type. Let's name to `atPath path`.

> `pathTransformWhereNilResultsWillBeEliminated`

This seems ill-fitting. I think it should be moved out of the function into the caller. That might mean flattening the result.

---

Here's what I'd suggest:

```swift
internal func list(treeish: String, atPath path: String, in repository: RepositoryURL) -> SignalProducer<String, CarthageError>
```